src/script/interpreter.h

```diff
@@ -215,6 +216,9 @@ struct ScriptExecutionData
     bool m_validation_weight_left_init = false;
     //! How much validation weight is left (decremented for every successful non-empty signature check).
     int64_t m_validation_weight_left;
+
+    //! The hash of the corresponding output
+    mutable std::optional<uint256> m_output_hash;
```

Comment:
Perhaps it is cleaner to not make this `mutable`, but instead make the places where `ScriptExecutionData` is used non-`const`? That'd make it also clear that it is not safe for concurrent modification.

Reply:
i think that seems reasonable, but it is a little bit of a tradeoff because you really ought not to be modifying the fields after creation except in this case, and non-consting it would make that unclear.

i'd also (ideally) make the m_output_hash retrieved through a getter function rather than inline logic, but I didn't want to make this PR more complex of an interface change.

maybe in a followup PR we could do some refactors to make it gotten through a Getter and rename the class to MutableScriptExecutionData? Or break out the Mutable and Immutable part as two separate caches? Also an option to just always pre-cache this, but that violates 'dont pay for what you dont use'

Reply:
It's a simple struct with some context data. As long as there is no benefit to actually having anything immutable, I don't think there is much of a problem with just making it non-const everywhere. `m_validation_weight_left` is also changed at runtime.